
Necessary fixes to build BFQ with:
"block-20160701-do-not-merge-requests-without-consulting-with-io-scheduler.patch"

diff -rup linux-4.4.16.old/block/bfq-iosched.c linux-4.4.16/block/bfq-iosched.c
--- linux-4.4.16.old/block/bfq-iosched.c	2016-07-27 19:53:29.375582782 +0200
+++ linux-4.4.16/block/bfq-iosched.c	2016-07-27 19:51:42.228059894 +0200
@@ -1527,7 +1527,7 @@ static int bfq_merge(struct request_queu
 	struct request *__rq;
 
 	__rq = bfq_find_rq_fmerge(bfqd, bio);
-	if (__rq && elv_rq_merge_ok(__rq, bio)) {
+	if (__rq && elv_bio_merge_ok(__rq, bio)) {
 		*req = __rq;
 		return ELEVATOR_FRONT_MERGE;
 	}
@@ -4774,7 +4774,7 @@ static struct elevator_type iosched_bfq
 #ifdef CONFIG_BFQ_GROUP_IOSCHED
 		.elevator_bio_merged_fn =	bfq_bio_merged,
 #endif
-		.elevator_allow_merge_fn =	bfq_allow_merge,
+		.elevator_allow_bio_merge_fn =	bfq_allow_merge,
 		.elevator_dispatch_fn =		bfq_dispatch_requests,
 		.elevator_add_req_fn =		bfq_insert_request,
 		.elevator_activate_req_fn =	bfq_activate_request,
